src/data.js

```diff
@@ -1,10 +1,12 @@
-export function fetchAllData(url) {
-  return fetch(url)
+export function fetchBreeds() {
+  return fetch("https://api.thedogapi.com/v1/breeds/")
     .then((response) => {
       if (response.ok) {
         return response.json();
       }
     })
 
-    .catch((error) => {});
+    .catch((error) => {
+      throw new Error(error.message);
```

Comment:
a če samo returnaš error -> `return error` ne dela? 
je potrebno "generirat" nov error z `throw new Error`?

pomoje je boljš da se cel error object pošlje v komponento in tam izpiše samo message -  `error.message`.. kdaj boš recimo mogu glede na tip errorja še kaj delat v komponenti..

Reply:
Popravil sem kodo tako, da returna samo error, vendar sem moral popraviti kodo v `fetchBreeds `fn v `BreedsList `komponenti, ker je `fetchBreeds `fn returnala error object kot resolved promise in sem moral dodati  še en kriterij  
```
if (results.message) {
            this.error = "Error: " + results.message;
          }
```
drugače mi ni delovalo pravilno.

Reply:
je to mogoče tako zaradi typo-ta v urlju? ker to da imaš nek "error" v thenu je mal čudn, ker bi se mogu promise rejectat in pol v catchu to ujet, če je error. 
Možno pa da je tak API, ne vem točno :D

Reply:
s tem typo-tam sem hotel stestirati catch error in mi je bilo čudno zakaj, ko catcha error v `fetchBreeds `v `data.js`, zakaj mi potem znotraj. `then((results)`... v `BreedsList.vue` vrže rezultat ta error. A ni tako,da bi morl tukaj catchat error in zalavfati ta 
` .catch((error) => {
          this.isLoading = false;
          this.error = error.message;
        });`

Reply:
odvisno kako ima API poštimano handlanje errorjev.. sepravi tukej če je error ti vrže v then in ne v catch?

Reply:
Tako ja. Nisem pa vedel,da je to odvisno od API.

Reply:
Wrapper,kot kaj? Kdaj, v katerem primeru, se bo potem ta drugi `catch ` ki ga imam v `BreedsList.vue` zalavfal? Če v `data.js`, tudi, če je error ga tako da v `then ` kot resolved promise? To mi ni ravno jasno, če sem iskren...